postgres/datadog_checks/postgres/statements.py

```diff
@@ -164,6 +164,18 @@ def collect_per_statement_metrics(self, db, db_version, tags):
             self._log.exception('Unable to collect statement metrics due to an error')
             return []
 
+    def _load_pg_stat_statements_safe(self, db, tags):
+        try:
+            self._load_pg_stat_statements(db)
+        except psycopg2.errors.ObjectNotInPrerequisiteState as e:
```

Comment:
While we're here there's a couple more errors that we should handle. The agent may not have permission to read it, or the user may have configured `pg_stat_statements_view` and there's something wrong with that view (missing, bad definition). 

Perhaps better to catch a generic database exception here and log that? We can include the type of the exception in the tags like we do for `dd.postgres.statement_samples.error` already.

Reply:
I added this in addition to keeping the specific handling for `pg_stat_statements_not_enabled`.